lib/EnzymeCore/src/EnzymeCore.jl

```diff
@@ -206,6 +207,46 @@ end
 @inline batch_size(::BatchMixedDuplicated{T,N}) where {T,N} = N
 @inline batch_size(::Type{BatchMixedDuplicated{T,N}}) where {T,N} = N
 
+"""
+    batchify_activity(::Type{A}, ::Val{B})
+
+Turn an activity (or [`Annotation`](@ref)) type `A` into the correct activity type for a batch of size `B`.
+
+# Examples
+
+```jldoctest
+julia> using EnzymeCore
+
+julia> EnzymeCore.batchify_activity(Active{Float64}, Val(2))
+Active{Float64}
+
+julia> EnzymeCore.batchify_activity(Duplicated{Vector{Float64}}, Val(2))
+BatchDuplicated{Vector{Float64}, 2}
+```
+"""
+batchify_activity(::Type{Active{T}}, ::Val{B}) where {T,B} = Active{T}
+batchify_activity(::Type{Duplicated{T}}, ::Val{B}) where {T,B} = BatchDuplicated{T,B}
```

Comment:
also does this need to go in EnzymeCore, or can this live in Enzyme proper atm (and perhaps later get upgraded to core as needed)

Reply:
The reason I put it in EnzymeCore is because it doesn't depend on anything in Enzyme proper, and deals purely with activities. To me that makes sense

Reply:
Now in Enzyme proper, along with `Seed` and `BatchSeed`